File: lib/depricated/php/benchmarks.php
<?php
include "db.php";

$database = new db("localhost", 3434, "admin", "password");

function tests($db){
    for ($i=0; $i < 1000; $i++) { 
        $path = "/".$i;
        echo $db->push($path, "data")."\n";
        echo $db->get($path, "data")."\n";
        echo $db->delete($path)."\n";
    }
}
$start_time = microtime(TRUE);
tests($database);
$end_time = microtime(TRUE);
/*
- 2.5969429016113
- 3.0023081302643
- 3.6546838283539
- 3.2196941375732
- 3.3984389305115
- 3.178200006485
- 3.2584710121155
*/
echo $end_time - $start_time."\n";
?>

File: lib/depricated/ruby/benchmarks.rb
require 'benchmark'
require_relative "db"

dab = DB.new "localhost", 3434, "admin", "password"

#print out everything
def test(db)
    for i in (0..1000).to_a do
        path = "/" + i.to_s
        puts db.push(path, "HELLO WORLD")
        puts db.get(path)
        puts db.delete(path)
    end
end

#dont print
def no(db)
    for i in (0..1000).to_a do
        path = "/" + i.to_s
        db.push(path, "HELLO WORLD")
        db.get(path)
        db.delete(path)
    end
end

#only print db.get
def re(db)
    for i in (0..1000).to_a do
        path = "/" + i.to_s
        db.push(path, "HELLO WORLD")
        puts db.get(path)
        db.delete(path)
    end
end


a = Benchmark.measure { test(dab) } # ("write/read/gets with puts:")   
b = Benchmark.measure { no(dab) } # ("write/read/gets with no puts:")
c = Benchmark.measure { re(dab) } # ("write/read/gets with puts on DB.get:") 

f = File.new("bench.txt", "a")
f.write("write/read/gets with puts:")
f.write(a)
f.write("write/read/gets with no puts:")
f.write(b)
f.write("write/read/gets with puts on DB.get:")
f.write(c)
f.close()


=begin
with puts
=end